torch/csrc/jit/pickler.cpp

```diff
@@ -126,6 +129,12 @@ void Pickler::pushIValueImpl(const IValue& ivalue) {
     err << ". Please define serialization methods via torch::jit::pickle_ for "
            "this class.";
     AT_ERROR(err.str());
+  } else if (ivalue.isRRef()) {
+    #ifdef USE_DISTRIBUTED
+      pushRRef(ivalue);
+    #else
+      TORCH_INTERNAL_ASSERT(false, "RRef pickling is only supported with the distributed package");
```

Comment:
Use TORCH_CHECK here or C10_THROW_ERROR. TORCH_INTERNAL_ASSERT will show up as a bug to the user and ask them to file a gh issue.

Also, what was the reason we needed to add this #ifdef? If USE_DISTRIBUTED is false, the user can't even create RRefs if I understand correctly, so how will they even pickle/unpickle one?

Reply:
synced up offline, USE_DISTRIBUTED is needed here as for some use cases, users do not need to compile rref_context.h if they are not building distributed package